File: 0070-climbing-stairs/0070-climbing-stairs.go
func climbStairs(n int) int {
    if n < 3 {
        return n;
    }

    var dp [3]int;
    dp[0] = 1;
    dp[1] = 2;
    for i := 2; i < n; i++ {
        dp[i%3] = dp[(i - 1)%3] + dp[(i - 2)%3];
    }

    return dp[(n - 1)%3];
}

File: 0070-climbing-stairs/0070-climbing-stairs.rs
use std::vec::Vec;

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n < 3 {
            return n;
        }

        let mut dp = vec![0;3];
        dp[0] = 1;
        dp[1] = 2;
        for i in 2..n {
            dp[(i%3) as usize] = dp[((i - 1)%3) as usize] + dp[((i - 2)%3) as usize];
        }

        return dp[((n - 1)%3) as usize];
    }
}